### PR TITLE
refactor: clean up unnecessary props and types, refactor `<Route>` to always render as children

### PR DESCRIPTION
**Pull request by emmenko**
Extracted from #2821

* The generic type argument `AdditionalEnvironmentProperties` is not really used and needed by the internal components. Let's remove it.
* Renamed some types for component props.
* Refactored `<Route>` components to render as children ([as recommended when upgrading to v6](https://reactrouter.com/en/main/upgrading/v5)).

### CODE REVIEW

File: packages/application-shell/src/components/application-entry-point/application-entry-point.tsx [hunk 1]
```diff
@@ -8,17 +8,15 @@ import { PageUnauthorized } from '@commercetools-frontend/application-components
 import { entryPointUriPathToPermissionKeys } from '@commercetools-frontend/application-config/ssr';
 import RouteCatchAll from '../route-catch-all';
 
-type Props<AdditionalEnvironmentProperties extends {}> = {
-  environment: TProviderProps<AdditionalEnvironmentProperties>['environment'];
+type TApplicationEntryPointProps = {
+  environment: TProviderProps<{}>['environment'];
```
Comment (emmenko) [inline packages/application-shell/src/components/application-entry-point/application-entry-point.tsx:12]: This type of change is done in many places in this PR.

* Renamed the type
* Removed the type argument `AdditionalEnvironmentProperties`

File: packages/application-shell/src/components/application-shell-provider/application-shell-provider.tsx [hunk 2]
```diff
@@ -24,73 +24,66 @@ import ErrorBoundary from '../error-boundary';
...
-    useCoercedEnvironmentValues<AdditionalEnvironmentProperties>(
-      props.environment
-    );
+  const coercedEnvironmentValues = useCoercedEnvironmentValues(
+    props.environment
+  );
   const browserLocale = getBrowserLocale(window);
   return (
-    <>
+    <Suspense fallback={<ApplicationLoader />}>
```
Comment (emmenko) [inline packages/application-shell/src/components/application-shell-provider/application-shell-provider.tsx:48]: Just moved up...

File: packages/application-shell/src/components/application-shell/application-shell.tsx [hunk 3]
```diff
@@ -461,10 +459,8 @@ export const RestrictedApplication = <
                                   }
                                 </Route>
                                 {/* Project routes */}
-                                <Route
-                                  exact={true}
-                                  path="/"
-                                  render={() => {
```
Comment (emmenko) [inline packages/application-shell/src/components/application-shell/application-shell.tsx:467]: No more `render` or `component` prop, routes are rendered as children of `<Route>`

File: packages/constants/src/constants.ts [hunk 4]
```diff
@@ -191,13 +191,9 @@ export interface ApplicationWindow extends Window {
     frontendHost: string;
     servedByProxy: boolean;
     // Optional properties. To use them, pass them to the `additionalEnv` object of the application config.
-    mcProxyApiUrl?: string;
```
Comment (emmenko) [inline packages/constants/src/constants.ts:194]: It's only used internally and should not be defined as a "default" field

File: packages/constants/src/constants.ts [hunk 5]
```diff
@@ -191,13 +191,9 @@ export interface ApplicationWindow extends Window {
...
     ldClientSideId?: string;
     trackingSentry?: string;
     trackingGtm?: string;
-    enableSignUp?: boolean;
-    enableLongLivedFeatureFlags?: boolean;
-    useFullRedirectsForLinks?: boolean;
```
Comment (emmenko) [inline packages/constants/src/constants.ts:200]: Same for these

File: packages/application-shell/src/hooks/use-applications-menu/use-applications-menu.ts [hunk 6]
```diff
@@ -30,6 +30,9 @@ export type Config = {
     onError?: QueryFunctionOptions['onError'];
   };
 };
+type TAdditionalEnvironmentProperties = {
+  mcProxyApiUrl?: string;
+};
```
Comment (emmenko) [inline packages/application-shell/src/hooks/use-applications-menu/use-applications-menu.ts:35]: We can then simply define a type to extend the app context and include the fields that we expect here to be optionally provided (via the [`additionalEnv`](https://docs.commercetools.com/custom-applications/api-reference/application-config#additionalenv) config field)

File: packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx [hunk 7]
```diff
@@ -81,11 +81,16 @@ const parseFlags = (fetchedFlags: TFetchedFlags): TParsedHttpAdapterFlags =>
     ])
   );
 
-export const SetupFlopFlipProvider = (props: Props) => {
+type TAdditionalEnvironmentProperties = {
+  enableLongLivedFeatureFlags?: boolean;
+};
```
Comment (emmenko) [inline packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx:86]: Here's another one

File: packages/application-shell/src/components/quick-access/quick-access.tsx [hunk 8]
```diff
@@ -117,7 +117,9 @@ const QuickAccess = (props: Props) => {
     customApplications: true,
     canViewDashboard: true,
   });
-  const applicationContext = useApplicationContext();
+  const applicationContext = useApplicationContext<{
+    useFullRedirectsForLinks?: boolean;
+  }>();
```
Comment (emmenko) [inline packages/application-shell/src/components/quick-access/quick-access.tsx:122]: Here as well

File: packages/application-shell/src/components/project-container/project-container.tsx [hunk 9]
```diff
@@ -25,14 +30,11 @@ import ApplicationEntryPoint from '../application-entry-point';
 import messages from './messages';
 
 type QueryParams = {
-  projectKey: string;
+  projectKey?: string;
```
Comment (emmenko) [inline packages/application-shell/src/components/project-container/project-container.tsx:33]: The query param is actually optional.

File: packages/application-shell/src/components/fetch-project/fetch-project.tsx [hunk 10]
```diff
@@ -15,13 +15,13 @@ type RenderFnArgs = {
   error?: ApolloError;
   project: TFetchProjectQuery['project'];
 };
-type Props = {
-  projectKey: string;
+type TFetchProjectProps = {
+  projectKey?: string;
```
Comment (emmenko) [inline packages/application-shell/src/components/fetch-project/fetch-project.tsx:19]: It's optional!

File: packages/application-shell/src/components/fetch-project/fetch-project.mc.graphql [hunk 11]
```diff
@@ -1,4 +1,4 @@
-query FetchProject($projectKey: String!) {
+query FetchProject($projectKey: String) {
```
Comment (emmenko) [inline packages/application-shell/src/components/fetch-project/fetch-project.mc.graphql:1]: Yep, it's optional 😊

File: packages/application-shell/src/components/navbar/navbar.tsx [hunk 12]
```diff
@@ -624,16 +624,16 @@ const NavBarLayout = forwardRef<HTMLElement, NavBarLayoutProps>(
 );
 NavBarLayout.displayName = 'NavBarLayout';
 
-type NavbarProps<AdditionalEnvironmentProperties extends {}> = {
+type NavbarProps = {
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/components/navbar/navbar.tsx:627]: **nit**
```suggestion
type TNavbarProps = {
```

**Review by kark — APPROVED** [review]
Thanks, it all looks good to me!

**Review by CarlosCortizasCT — APPROVED** [review]
👍

**Pull request merged**